WolvenKit.RED4/Types/ClassesExt/Appendix/CMaterialInstance.cs

```diff
@@ -8,13 +8,15 @@ public partial class CMaterialInstance : IRedAppendix
     [REDProperty(IsIgnored = true)]
     public CArray<CKeyValuePair> Values
     {
-        get => GetPropertyValue<CArray<CKeyValuePair>>();
+        get => GetPropertyValue<CArray<CKeyValuePair>>()!; // set in PostConstruct, so not nullable
         set => SetPropertyValue<CArray<CKeyValuePair>>(value);
+
     }
 
     partial void PostConstruct()
     {
-        Values = new CArray<CKeyValuePair>();
+        Values = [];
+        Metadata ??= new SerializationDeferredDataBuffer();
```

Comment:
Why this change?

Reply:
Type checking. The property is not nullable, but it's not necessary initialized, and I actually ran into an exception about this.

Reply:
General Info (not specialy for `CMaterialInstance`)

Changing the "default" value from `null` to `new()` indeed changes how the CR2W is written.
It compares the set value to the default one and if it is default, the property doesn't get written at all.

So the logic changes from:
If it's null, it doesn't get written
If it's a buffer (even a empty one) it gets written.

To:
If it's null, it would try to write that (don't know what would happen then currently)
If it's a empty buffer, it won't get written
If it's a filled buffer, it gets written.

So in theory such things can break in engine. 

If these warning annoys you, should probably disable them for `ClassesExt` via `.editorconfig` like in `Classes`

Reply:
OK, I get that it has side effects. My dilemma here is that I kind of want these properties to be initialized inside of WKit, because without this change I am constantly initializing them by hand (every time after creating a new material e.g.)

Should I define them in the initialization again, or what would be the engine-proof solution here?

Reply:
Easiest (without changing the logic) would probably be creating a factory (helper) for them (i.e in .Common / .ModKit)?
Which yields you the classes as you need them the most?

Reply:
For such cases like creating new materials you do indeed need to create new objects yeah, because null here is a valid value for the cr2w

Reply:
OK, I'll see that I try and create a factory here then